estimation.py

```diff
@@ -0,0 +1,198 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import contextlib
+import gc
+import os
+
+import torch
+import torch.nn.functional as F
+from torch._guards import active_fake_mode
+from torch._subclasses.fake_tensor import FakeTensorMode
+from torch.distributed import destroy_process_group
+from torch.distributed._tools.fsdp2_mem_tracker import FSDPMemTracker
+from torch.distributed.tensor.parallel import loss_parallel
+from torch.testing._internal.distributed.fake_pg import FakeStore
+
+from torchtitan.config_manager import JobConfig
+from torchtitan.datasets import create_tokenizer
+from torchtitan.float8_linear import build_fp8_linear
+from torchtitan.logging_utils import init_logger, logger
+from torchtitan.lr_scheduling import get_lr_scheduler
+from torchtitan.models import model_name_to_cls, model_name_to_tokenizer, models_config
+from torchtitan.parallelisms import models_parallelize_fns, ParallelDims
+from train import build_optimizer
+
+
+def estimate_memory(job_config: JobConfig):
+    init_logger()
+    logger.info("Estimating memory usage...")
+    gc.disable()
+    gc.collect(1)
+
+    # Get the world size
+    world_size = int(os.environ["WORLD_SIZE"])
+
+    # if tp > or pp > 1, we exit
+    if (
+        job_config.training.tensor_parallel_degree > 1
+        or job_config.experimental.pipeline_parallel_degree > 1
+    ):
+        logger.info(
+            "Tensor parallelism and pipeline parallelism are not supported yet."
+        )
+        return
+
+    # fake tensor doesn't work with fused rmsnorm
+    if (
+        job_config.model.norm_type == "fused_rmsnorm"
+        and job_config.estimate.mode == "fake"
+    ):
+        logger.info("Fused RMSNorm is not supported yet under fake estimation mode.")
+        return
```

Comment:
perhaps just inform them you are updating and go ahead and switch to RMSNorm directly instead of failing out and making them go change it?
i.e. job_config.model.norm_type="rmsnorm"
and proceed?

Reply:
Makes sense!